src/coreclr/tools/Common/TypeSystem/IL/Stubs/ComparerIntrinsics.cs

```diff
@@ -85,6 +85,10 @@ private static InstantiatedType GetComparerForType(TypeDesc type, string flavor,
                 // The comparer will be determined at runtime. We can't tell the exact type at compile time.
                 return null;
             }
+            else if (flavor == "EqualityComparer" && context.SystemModule.GetKnownType("System", "Byte").Equals(type))
+            {
+                return context.SystemModule.GetKnownType("System.Collections.Generic", $"ByteEqualityComparer");
+            }
```

Comment:
@MichalStrehovsky Is there any way to simplify this?

Reply:
https://github.com/dotnet/runtime/pull/88006/files#r1241212332